ouroboros-network/src/Ouroboros/Network/Diffusion.hs

```diff
@@ -302,24 +319,30 @@ runDataDiffusion tracers
                Left sd -> do
                    a <- Socket.getSocketName sd
                    case a of
-                        (Socket.SockAddrUnix path) ->
+                        (Socket.SockAddrUnix path) -> do
+                          traceWith dtDiffusionInitializationTracer $ DiffusionInitiatilizationCreatingLocalServerSocketForUnixPath path
                           return (sd, Snocket.localSnocket iocp path)
-                        _                          ->
+                        unsupportSocket -> do
                             -- TODO: This should be logged.
+                            traceWith dtDiffusionInitializationTracer $ DiffusionInitiatilizationUnsupportedSocketType (show unsupportSocket)
                             throwIO UnsupportedLocalSocketType
 #endif
-               Right a -> do
-                   let sn = Snocket.localSnocket iocp a
-                   sd <- Snocket.open sn (Snocket.addrFamily sn $ Snocket.localAddressFromPath a)
+               Right path -> do
+                   let sn = Snocket.localSnocket iocp path
+                   traceWith dtDiffusionInitializationTracer $ DiffusionInitiatilizationCreateLocalServerSocketForSnocketPath path
+                   sd <- Snocket.open sn (Snocket.addrFamily sn $ Snocket.localAddressFromPath path)
+                   traceWith dtDiffusionInitializationTracer $ DiffusionInitiatilizationCreatedLocalServerSocketForSnocketPath path
                    return (sd, sn)
         )
         (\(sd,sn) -> Snocket.close sn sd) -- We close the socket here, even if it was provided for us.
         (\(sd,sn) -> do
 
           case daLocalAddress of
                Left _ -> pure () -- If a socket was provided it should be ready to accept
-               Right a -> do
-                 Snocket.bind sn sd $ Snocket.localAddressFromPath a
+               Right path -> do
+                 traceWith dtDiffusionInitializationTracer $ DiffusionInitiatilizationBindingToSocket path (show sd)
```

Comment:
```suggestion
                 traceWith dtDiffusionInitializationTracer $ BindingSocket (show sd) path
```

by including socket first it will be more natural to read the log message `BindingSocket <socket: 13> /some/path`